datadog/resource_datadog_user.go

```diff
@@ -79,79 +81,94 @@ func resourceDatadogUserExists(d *schema.ResourceData, meta interface{}) (b bool
 }
 
 func resourceDatadogUserCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*datadog.Client)
-
-	var u datadog.User
-	u.SetDisabled(d.Get("disabled").(bool))
-	u.SetEmail(d.Get("email").(string))
-	u.SetHandle(d.Get("handle").(string))
-	u.SetIsAdmin(d.Get("is_admin").(bool))
-	u.SetName(d.Get("name").(string))
-	u.SetAccessRole(d.Get("access_role").(string))
+	providerConf := meta.(*ProviderConfiguration)
+	client := providerConf.DatadogClientV1
+	auth := providerConf.Auth
+
+	var userCreate datadog.User
+	userCreate.SetEmail(d.Get("email").(string))
+	userCreate.SetHandle(d.Get("handle").(string))
+	userCreate.SetName(d.Get("name").(string))
+	userCreate.SetAccessRole(datadog.AccessRole(d.Get("access_role").(string)))
+
+	var userUpdate datadog.User
+	userUpdate.SetEmail(d.Get("email").(string))
+	userUpdate.SetName(d.Get("name").(string))
+	userUpdate.SetAccessRole(datadog.AccessRole(d.Get("access_role").(string)))
+	userUpdate.SetDisabled(d.Get("disabled").(bool))
 
 	// Datadog does not actually delete users, so CreateUser might return a 409.
 	// We ignore that case and proceed, likely re-enabling the user.
-	if _, err := client.CreateUser(u.Handle, u.Name); err != nil {
+	if _, _, err := client.UsersApi.CreateUser(auth).Body(userCreate).Execute(); err != nil {
 		if !strings.Contains(err.Error(), "API error 409 Conflict") {
```

Comment:
let's instead check for the status code directly:
```suggestion
	if _, httpresp, err := client.UsersApi.CreateUser(auth).Body(userCreate).Execute(); err != nil {
		if httpresp.StatusCode != 409 {
```